lib/mayaUsd/fileio/primUpdaterManager.cpp

```diff
@@ -533,31 +540,21 @@ SdfPath getDstSdfPath(const Ufe::Path& ufePulledPath, const SdfPath& srcSdfPath,
 
 //------------------------------------------------------------------------------
 //
+// Create an updater for use with both pushCustomize() traversals /
+// customization points: pushCopySpec() and pushEnd().
+//
 UsdMayaPrimUpdaterSharedPtr createUpdater(
-    const Ufe::Path&                 ufePulledPath,
     const SdfLayerRefPtr&            srcLayer,
     const SdfPath&                   srcPath,
-    const SdfLayerRefPtr&            dstLayer,
     const SdfPath&                   dstPath,
     const UsdMayaPrimUpdaterContext& context)
 {
-    // The root of the pulled hierarchy is crucial for determining push
-    // behavior.  When pulling, we may have created a Maya pull hierarchy root
-    // node whose type does not map to the same prim updater as the original
-    // USD prim, i.e. multiple USD prim types can map to the same pulled Maya
-    // node type (e.g. transform, which is the fallback Maya node type for many
-    // USD prim types).  Therefore, if we're at the root of the src hierarchy,
-    // use the prim at the pulled path to create the prim updater; this will
-    // occur on push, when the srcPath is in the temporary layer.
-    const bool usePulledPrim = (srcPath.GetPathElementCount() == 1);
-
     auto primSpec = srcLayer->GetPrimAtPath(srcPath);
     if (!TF_VERIFY(primSpec)) {
         return nullptr;
     }
 
-    TfToken typeName = usePulledPrim ? MayaUsd::ufe::ufePathToPrim(ufePulledPath).GetTypeName()
-                                     : primSpec->GetTypeName();
+    auto typeName = primSpec->GetTypeName();
```

Comment:
Another crucial improvement: because the pull / push round trip is now USD type-preserving, we can now simply use the USD type when creating updaters during the pushCopySpecs() and pushEnd() temporary stage traversals.

Reply:
Trying to use the pulled prim was buggy, because in the pushEnd() traversal the Maya reference prim no longer exists when using the cache to variant workflow!  It has been replaced by the cache variant.